test/libsolidity/syntaxTests/license/license_whitespace_1.sol

```diff
@@ -0,0 +1,2 @@
+//     SPDX-License-Identifier: GPL-3.0
+contract C {}
```

Comment:
One more possible test case:
```solidity
// SPDX-License-Identifier    : GPL-3.0
contract C {}
```

Reply:
Also more test ideas:
- Whole comment inside Unicode direction override marks (`/// <LRO>SPDX-License-Identifier: GPL-3.0<PDF>`)
- Missing colon (`/// SPDX-License-Identifier GPL-3.0`)
- Multi-line NatSpec comment (`/** SPDX-License-Identifier GPL-3.0 */`)
- Multi-line comment (`/* SPDX-License-Identifier GPL-3.0 */`)
- License comment at the bottom of the file
- License comment ending with a Mac or Windows newline
- License comment on a non-empty line (`pragma solidity ^0.8.0; // SPDX-License-Identifier: GPL-3.0`)
- License comment in a string (`import "// SPDX-License-Identifier: GPL-3.0"`)
- No whitespace at all (`//SPDX-License-Identifier:GPL-3.0`, `/*SPDX-License-Identifier:GPL-3.0*/`)

Reply:
If you have the time please feel free to add them.

Reply:
Not today unfortunately and I'll be off since Monday (though I'll probably work a bit over the weekend just to finally finish the stuff with binaries). If you're fine putting this on hold until I'm back, feel free to assign the PR to me. Otherwise I can approve the PR even without them once everything else is fixed. The tests you added already some value so no reason not to merge it just because we can add even more :)

Reply:
I have now added all these test cases. Both as `.sol` files and as metadata tests to see if the license is actually picked up.

There actually was a bug in the code and `import "// SPDX-License-Identifier: GPL-3.0"` detected it - `SyntaxTest.cpp` not only adds the preamble but also subtracts the added characters from error locations. This second part was still assuming the old preamble which resulted in wrong locations. I initially thought that it's a bug on `develop` in the code that reports the error but after some debugging I realized that it's only code from this PR that messes things up.